### ISSUE TITLE
Object Spread should not be preferred when the argument is this.

### ISSUE THREAD

**Issue by snood1205** [palantir/tslint#3117]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.3.3
- __Running TSLint via__: WebStorm

#### TypeScript code being linted

```ts
const form = Object.assign({}, this);
```

with `tslint.json` configuration:

```json
{
    "prefer-object-spread": true
}
```

#### Actual behavior

TSLint gives me the message

> Use the object spread operator instead. (prefer-object-spread)

#### Expected behavior

TSLint should be ok with this usage of `Object.assign({}, this)` because per TS2698, "spread types may only be created from object types."

Somewhat related to this is: #2824

**Comment by ajafff** [comment]
Have you already searched for or opened a corresponding issue in the typescript repo? 
I think this is better fixed  in the typescript compiler if they consider this a bug. Otherwise we can add an exception for this case to the rules.

**Comment by snood1205** [comment]
@ajafff The related TS issues are [https://github.com/Microsoft/TypeScript/issues/10727](#10727) and [https://github.com/Microsoft/TypeScript/issues/12756](#12756). It seems from the latter that people want support for this but it doesn't really seem to have been fixed thus far at least? I'm not sure if it ever will be fixed.

**Comment by ajafff** [comment]
The linked PR that adds spread types to typescript looks abandoned. It's reasonable to add an exception to the rule until the compiler supports it.

**Comment by ajafff** [comment]
Fixed by #3126

**Comment by snood1205** [comment]
@ajafff Awesome, thank you!
